### PR TITLE
raftstore: support rocksdb cf.

### PR DESCRIPTION
**Pull request by disksing**
@BusyJay @siddontang @ngaut @zhangjinpeng1987

### CODE REVIEW

File: src/raftstore/store/engine.rs [hunk 1]
```diff
@@ -141,6 +162,12 @@ impl Peekable for DB {
         let v = try!(self.get(key));
         Ok(v)
     }
+    fn get_value_cf(&self, cf: &str, key: &[u8]) -> Result<Option<DBVector>> {
```
Comment (BusyJay) [inline src/raftstore/store/engine.rs]: Any test cover this?

File: src/raftstore/store/peer.rs [hunk 2]
```diff
@@ -1196,6 +1202,10 @@ impl Peer {
         Ok(resp)
     }
 
+    fn get_cf_handle(&self, cf: &str) -> Result<&DBCFHandle> {
+        self.engine.cf_handle(cf).ok_or_else(|| Error::RocksDb("cf not found".to_string()))
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: add the cf name in error .

File: src/raftstore/store/engine.rs (hunk 1 above)
Comment (disksing) [inline src/raftstore/store/engine.rs]: There are some simple tests in `tests/storage/test_raftkv.rs`

Comment (BusyJay) [inline src/raftstore/store/engine.rs]: I think it's better to enhance the test cases in this file.

**Comment by disksing** [comment]
PTAL @BusyJay @siddontang

File: src/raftstore/store/engine.rs [hunk 3]
```diff
@@ -50,15 +50,36 @@ impl Drop for Snapshot {
...
+    let opts = Options::new();
+    new_engine_opt(opts, path, cfs)
+}
+
+pub fn new_engine_opt(mut opts: Options, path: &str, cfs: &[&str]) -> Result<Arc<DB>> {
+    // TODO: configurable opts for each CF.
+    opts.create_if_missing(false);
+    match DB::open_cf(&opts, path, cfs) {
+        Ok(db) => return Ok(Arc::new(db)),
+        Err(e) => warn!("open rocksdb fail: {}", e),
```
Comment (siddontang) [inline src/raftstore/store/engine.rs]: how about origin has cf a, b but we pass here a, b, c ?

File: src/raftstore/store/engine.rs [hunk 4]
```diff
@@ -158,6 +185,17 @@ impl Peekable for Snapshot {
         let v = try!(self.db.get_opt(key, &opt));
         Ok(v)
     }
+    fn get_value_cf(&self, cf: &str, key: &[u8]) -> Result<Option<DBVector>> {
```
Comment (siddontang) [inline src/raftstore/store/engine.rs]: add a blank

File: src/raftstore/store/engine.rs (hunk 3 above)
Comment (disksing) [inline src/raftstore/store/engine.rs]: Not support now. Forgot to add a TODO here.

**Comment by disksing** [comment]
PTAL

**Comment by siddontang** [comment]
LGTM

**Comment by disksing** [comment]
@BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**